appleauth/auth_source.go

```diff
@@ -95,7 +95,7 @@ func (*ConnectionAppleIDSource) Fetch(conn *devportalservice.AppleDeveloperConne
 			Username:            conn.AppleIDConnection.AppleID,
 			Password:            conn.AppleIDConnection.Password,
 			Session:             "",
-			AppSpecificPassword: inputs.AppSpecificPassword,
+			AppSpecificPassword: appSpecificPassword(conn.AppleIDConnection, inputs.AppSpecificPassword),
```

Comment:
In my opinion, if using connection, then the connection app specific password should have priority, if defined (not nil, empty or whitespace only).
In case of the sources related to step inputs (InputAppleIDSource, InputAppleIDFastlaneSource) the connection should be ignored and the app specific password from the inputs used, as these sources are explicitly using the inputs to configure the parameters.